index.d.ts

```diff
@@ -0,0 +1,60 @@
+export interface Action {
+  type: string;
+}
+
+export type Reducer<S> = (state: S, action: Action) => S;
+
+export type Dispatch = (action: any) => any;
+
+export interface MiddlewareAPI<S> {
+  dispatch: Dispatch;
+  getState: () => S;
+}
+
+export interface Middleware {
+  <S>(api: MiddlewareAPI<S>): (next: Dispatch) => Dispatch;
+}
+
+export interface Store<S> {
+  dispatch: Dispatch;
+  getState: () => S;
+  subscribe: (listener: () => void) => () => void;
+  replaceReducer: (reducer: Reducer<S>) => void;
+}
+
+export interface StoreCreator<S> {
+  (reducer: Reducer<S>): Store<S>;
+  (reducer: Reducer<S>, enhancer: StoreEnhancer): Store<S>;
+  (reducer: Reducer<S>, initialState: S): Store<S>;
+  (reducer: Reducer<S>, initialState: S, enhancer: StoreEnhancer): Store<S>;
+}
+
+export type StoreEnhancer = <S>(next: StoreCreator<S>) => StoreCreator<S>;
+
+export const createStore: StoreCreator;
+
+export function combineReducers<S>(reducers: {[key: string]: Reducer<any>}): Reducer<S>;
+export function applyMiddleware<S>(...middlewares: Middleware[]): StoreEnhancer;
+
+
+export interface ActionCreator {
+  (...args: any[]): any;
```

Comment:
Once more, action creator here is anything that can be bound to `Dispatch`, so its return type is anything that `Dispatch` can accept.